src/app/features/charts/components/median-barchart/median-barchart.component.ts

```diff
@@ -0,0 +1,260 @@
+// -------------------------------------------------------------------------- //
+// External
+// -------------------------------------------------------------------------- //
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  HostListener,
+  Input,
+  OnChanges,
+  OnDestroy,
+  SimpleChanges,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import * as d3 from 'd3';
+
+// -------------------------------------------------------------------------- //
+// Internal
+// -------------------------------------------------------------------------- //
+import { MedianExpression } from '../../../../models';
+import { HelperService } from '../../../../core/services';
+
+// -------------------------------------------------------------------------- //
+// Component
+// -------------------------------------------------------------------------- //
+@Component({
+  selector: 'median-barchart',
+  templateUrl: './median-barchart.component.html',
+  styleUrls: ['./median-barchart.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class MedianBarChartComponent
+  implements OnChanges, AfterViewInit, OnDestroy
+{
+  private chartInitialized = false;
+  private tooltipInitialized = false;
+  private _data: MedianExpression[] = [];
+  private chart!: d3.Selection<any, unknown, null, undefined>;
+  private tooltip!: d3.Selection<any, unknown, null, undefined>;
+  private MEANINGFUL_EXPRESSION_THRESHOLD = Math.log2(5);
+  private maxValueY = -1;
+
+  get data() {
+    return this._data;
+  }
+  @Input() set data(data: MedianExpression[]) {
+    this._data = data
+      .filter((el) => el.medianlogcpm && el.medianlogcpm > 0)
+      .sort((a, b) => a.tissue.localeCompare(b.tissue));
+    this.maxValueY = Math.max(
+      this.MEANINGFUL_EXPRESSION_THRESHOLD,
+      d3.max(this._data, (d) => d.medianlogcpm) || 0
+    );
```

Comment:
Calculate a max value for the y-axis, so that the meaningful expression threshold is still shown if values are very small